apps/verticalhome/js/pin_apps.js

```diff
@@ -113,8 +157,137 @@
+        }
+        navigator.getDataStores(this.STORE_NAME).then(stores => {
+          this._storeRef = stores[0];
+          //TODO: need to iterate through obtained store, to read initial values for notifications.
+          return resolve(this._storeRef);
+        }, err => {
+          this.debug('ERROR '+ JSON.stringify(err));
+          return reject(err);
+        });
+      });
+    },
+
   };
 
   exports.PinAppManager = PinAppManager;
+
+  function LoadAllNotifications(store) {
+    return new Promise( (resolve, reject) => {
+      var storedNotifications = {};
+      var cursor = store.sync();
+      function cursorResolve(task) {
+        switch (task.operation) {
+          case 'update':
+          case 'add':
+            console.log('[el] got ADD/UPD task from cursor');
+            console.log('[el] ADD/UPD | task ' + JSON.stringify(task));
+            storedNotifications[task.id] = task.data;
+            break;
+
+          case 'remove':
+            console.log('[el] got REMOVE task from cursor');
+            delete storedNotifications[task.id];
+            break;
+
+          case 'clear':
+            console.log('[el] got CLEAR task from cursor');
+            storedNotifications = {};
+            break;
+
+          case 'done':
+            console.log('[el] got DONE task| returning = ' + JSON.stringify(storedNotifications));
+            resolve(storedNotifications);
+            return;
+        }
+
+        cursor.next().then(cursorResolve, reject);
+      };
+      cursor.next().then(cursorResolve, reject);
+
+    });
+  }
+
+  function CleanAllBubbles() {
+    app.getPinAppList().forEach( app => {
+      ShowNotifBubble(app.manifestURL, 0);
+    });
+  };
+
+  function ShowNotifBubble(appId,notifCount){
+
+      var els = document.getElementsByTagName('div');
+      var i = 0;
+
+      for (i = 0; i < els.length; i++) {
+        if (els[i].hasAttribute('data-manifesturl')) {
+          if (els[i].getAttribute('data-manifesturl') == appId) {
+
+            var unreadNotif = els[i].getElementsByClassName('unread_notif')[0];
+
+            /* start test part */
+//            var notifCountBase = parseInt(unreadNotif.innerHTML || 0);
+//            notifCountBase += notifCount;
+
+//            if(notifCountBase > 999){
+//              notifCountBase = 0;
+//            }
+            /* end test part */
+
+            /* uncomment for real life */
+             var notifCountBase;
+             notifCountBase = parseInt(notifCount);
+            /* /uncomment for real life */
+
+            if(notifCountBase > 0 || notifCountBase){
+              unreadNotif.style.display = "block";
+              unreadNotif.innerHTML = notifCountBase;
+            }
+            else{
+              unreadNotif.style.display = "none";
+              unreadNotif.innerHTML = 0;
+            }
+          }
+
+        }
+      }
+  };
+
+window.addEventListener("keydown", function(e){
+
+  switch(e.keyCode){
+    case 49:
```

Comment:
this is one more 'to be killed' part, it was created just for testing purposes (to simulate updates of app's counters) while real DataStore usage was in development.